core/modules/user/js/user.js

```diff
@@ -166,9 +174,23 @@ Backdrop.evaluatePasswordStrength = function (password, username) {
   // its length. Again, adapted from zxcvbn.
   strength = (Math.log(cardinality) / Math.log(2)) * password.length + 1;
 
-  // Check if password is the same as the username.
-  if (password !== '' && password.toLowerCase() === username.toLowerCase()) {
-    strength = 5;
+  // Check if password is the same as the username or email.
+  if (password !== '') {
+    if (password === username || password === email) {
```

Comment:
This was by intention. Password validation won't consider `myusername` and `MyUsErNaMe` the same, so I see no reason, why the indicator should.

Reply:
I'm still confused by this, but my head's not letting me understand it. Perhaps someone else can approve this change...?

Reply:
I say we leave in `.toLowerCase()` for username for backwards compatibility and ensure the PHP validation also matches this. I also think we should apply that to emails as well to be consistent with usernames.

Reply:
Case insensitive comparison is back.

Reply:
@herbdool this way the case-insensitive comparison is applied only if constrains are active (`.toLowerCase()` conversion only happens when `config.case_insensitive_comparison === 1`). We should move the conditional with the `.toLowerCase()` conversion before the `if (password === username || password === email)`, so that it is applied regardless of the setting being "Enforce strong passwords" or "Indicate weak passwords".

Reply:
@herbdool this way the case-insensitive comparison is applied only if constrains are active (`.toLowerCase()` conversion only happens when `config.case_insensitive_comparison === 1`). We should move the conditional with the `.toLowerCase()` conversion before the `if (password === username || password === email)`, so that it is applied regardless of the setting being "Enforce strong passwords" or "Indicate weak passwords":

edit: something funky happening with code suggestions when there are previous comments/reviews 🤔 ...let me try by suggesting how this entire block of code should be instead:

```php

```

Reply:
@klonos I agree with most of what you're saying. For backwards compatibility the comparison needs to be case insensitive even if constraints aren't being enforced. But the way we've got it now, `case_insensitive_comparison` is an odd toggle: it's the only one that allows for weaker passwords compared to before this PR, if an admin unselects it. And it's hidden behind the constraints toggle so it doesn't feel right to use it outside of that logic.

So I'm thinking that we have a three choices. Either we leave it as is and have the password indicator a little bit less strict for sites which are not enforcing constraints. Or use `case_insensitive_comparison` in the logic even if someone hasn't chosen to enforce password constraints (a bit sneaky). Or rethink this a bit. Perhaps we could have the constraints as options for the password strength indicator even if it's not enforced. I've got it working that way locally.

Reply:
>Perhaps we could have the constraints as options for the password strength indicator even if it's not enforced.

Yes, that's what I was going for.